Index: ft2demos-2.6.3/src/ftcommon.c
===================================================================
--- ft2demos-2.6.3.orig/src/ftcommon.c
+++ ft2demos-2.6.3/src/ftcommon.c
@@ -439,6 +439,7 @@
 
           if ( file == NULL )  /* shouldn't happen */
           {
+            free( (void *)font->filepathname );
             free( font );
             return FT_Err_Invalid_Argument;
           }
Index: ft2demos-2.6.3/src/ftdiff.c
===================================================================
--- ft2demos-2.6.3.orig/src/ftdiff.c
+++ ft2demos-2.6.3/src/ftdiff.c
@@ -1502,7 +1502,13 @@
 
         if ( text != NULL )
         {
-          fread( text, (unsigned int)tsize, 1, tfile );
+          if (fread( text, (unsigned int)tsize, 1, tfile ) < 1) {
+            fprintf( stderr, "could not read '%s'\n", textfile );
+            free( text );
+            text = (char *)default_text;
+          } else {
+            text[tsize] = 0;
+          }
           text[tsize] = 0;
         }
         else
