From e55a95a8a3f1f6fd216a1e7c08fdba8210d0a891 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michal=20=C4=8Ciha=C5=99?= <michal@cihar.com>
Date: Tue, 6 Dec 2016 10:21:02 +0100
Subject: [PATCH 1/1] Revert to auto for further tests
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Michal Čihař <michal@cihar.com>
---
 test/classes/ConfigTest.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/classes/ConfigTest.php b/test/classes/ConfigTest.php
index b15d1291e8..cd808340d5 100644
--- a/test/classes/ConfigTest.php
+++ b/test/classes/ConfigTest.php
@@ -329,7 +329,7 @@ class ConfigTest extends PMATestCase
         $this->object->checkGd2();
         $this->assertEquals(0, $this->object->get('PMA_IS_GD2'));
 
-        $this->object->set('GD2Available', $prevIsGb2Val);
+        $this->object->set('GD2Available', 'auto');
 
         if (!@function_exists('imagecreatetruecolor')) {
             $this->object->checkGd2();
-- 
2.11.0

