Subject: Adjust phpMyAdmin setup message to match our changes
From: Michal Čihař <nijel@debian.org>
Forwarded: not-needed

--- a/setup/frames/index.inc.php
+++ b/setup/frames/index.inc.php
@@ -119,9 +119,7 @@
         'notice', uniqid('config_saved'), __('Configuration saved.'),
         PMA_sanitize(
             __(
-                'Configuration saved to file config/config.inc.php in phpMyAdmin '
-                . 'top level directory, copy it to top level one and delete '
-                . 'directory config to use it.'
+                'Configuration saved to file /var/lib/phpmyadmin/config.inc.php.'
             )
         )
     );
