Index: handbrake-0.9.7/libhb/muxmkv.c
===================================================================
--- handbrake-0.9.7.orig/libhb/muxmkv.c	2012-07-16 09:40:20.928615427 +0200
+++ handbrake-0.9.7/libhb/muxmkv.c	2012-07-16 09:45:23.845291916 +0200
@@ -290,8 +290,8 @@
         {
             // For HE-AAC, write outputSamplingFreq too
             // samplingFreq is half of outputSamplingFreq
-            track->extra.audio.outputSamplingFreq = (float)audio->config.out.samplerate;
-            track->extra.audio.samplingFreq = track->extra.audio.outputSamplingFreq / 2.;
+//            track->extra.audio.outputSamplingFreq = (float)audio->config.out.samplerate;
+//            track->extra.audio.samplingFreq = track->extra.audio.outputSamplingFreq / 2.;
         }
         else
         {
