Description: Ignore missing portmidi/porttime build dependencies on kfreebsd
Forwarded: no
Author: A Mennucc1 <mennucc1@debian.org>
Last-Update: 2012-08-26

--- a/config_unix.py
+++ b/config_unix.py
@@ -222,7 +222,11 @@
 
     for d in DEPS[1:]:
         if not d.found:
-            if "-auto" not in sys.argv and not confirm("""
+            if ( d.name=='PORTMIDI' or d.name=='PORTTIME') and  \
+                    os.getenv('DEB_HOST_ARCH_OS') != 'linux':
+                print 'Debian specific hack: ignore missing dependency %s on %s .' % \
+                    (d.name , os.getenv('DEB_HOST_ARCH_OS'))
+            elif not confirm("""
 Warning, some of the pygame dependencies were not found. Pygame can still
 compile and install, but games that depend on those missing dependencies
 will not run. Would you like to continue the configuration?"""):
