Index: fontconfig-2.10.2/conf.d/40-nonlatin.conf
===================================================================
--- fontconfig-2.10.2.orig/conf.d/40-nonlatin.conf	2012-07-24 04:01:32.000000000 +0200
+++ fontconfig-2.10.2/conf.d/40-nonlatin.conf	2012-11-26 18:22:41.488844279 +0100
@@ -161,6 +161,10 @@
 		<default><family>sans-serif</family></default>
 	</alias>
 	<alias>
+		<family>MgOpen Moderna</family>
+		<default><family>sans-serif</family></default>
+	</alias>
+	<alias>
 		<family>Sazanami Gothic</family>
 		<default><family>sans-serif</family></default>
 	</alias>
Index: fontconfig-2.10.2/conf.d/60-latin.conf
===================================================================
--- fontconfig-2.10.2.orig/conf.d/60-latin.conf	2012-02-20 04:49:18.000000000 +0100
+++ fontconfig-2.10.2/conf.d/60-latin.conf	2012-11-26 18:22:38.608844139 +0100
@@ -4,8 +4,8 @@
 	<alias>
 		<family>serif</family>
 		<prefer>
-			<family>Bitstream Vera Serif</family>
 			<family>DejaVu Serif</family>
+			<family>Bitstream Vera Serif</family>
 			<family>Times New Roman</family>
 			<family>Thorndale AMT</family>
 			<family>Luxi Serif</family>
@@ -16,8 +16,8 @@
 	<alias>
 		<family>sans-serif</family>
 		<prefer>
-			<family>Bitstream Vera Sans</family>
 			<family>DejaVu Sans</family>
+			<family>Bitstream Vera Sans</family>
 			<family>Verdana</family>
 			<family>Arial</family>
 			<family>Albany AMT</family>
@@ -32,8 +32,8 @@
 	<alias>
 		<family>monospace</family>
 		<prefer>
-			<family>Bitstream Vera Sans Mono</family>
 			<family>DejaVu Sans Mono</family>
+			<family>Bitstream Vera Sans Mono</family>
 			<family>Inconsolata</family>
 			<family>Andale Mono</family>
 			<family>Courier New</family>
Index: fontconfig-2.10.2/conf.d/Makefile.am
===================================================================
--- fontconfig-2.10.2.orig/conf.d/Makefile.am	2012-10-30 01:53:37.000000000 +0100
+++ fontconfig-2.10.2/conf.d/Makefile.am	2012-11-26 18:22:38.608844139 +0100
@@ -25,6 +25,9 @@
 	README
 
 CONF_LINKS = \
+	10-antialias.conf \
+	10-hinting.conf \
+	10-hinting-slight.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
@@ -47,7 +50,12 @@
 
 templatedir = $(TEMPLATEDIR)
 template_DATA =				\
+	10-antialias.conf		\
 	10-autohint.conf		\
+	10-hinting.conf		\
+	10-hinting-full.conf		\
+	10-hinting-medium.conf		\
+	10-hinting-slight.conf		\
 	10-no-sub-pixel.conf		\
 	10-sub-pixel-bgr.conf		\
 	10-sub-pixel-rgb.conf		\
@@ -66,6 +74,7 @@
 	49-sansserif.conf		\
 	50-user.conf			\
 	51-local.conf			\
+	53-monospace-lcd-filter.conf			\
 	60-latin.conf			\
 	65-fonts-persian.conf		\
 	65-khmer.conf			\
Index: fontconfig-2.10.2/conf.d/Makefile.in
===================================================================
--- fontconfig-2.10.2.orig/conf.d/Makefile.in	2012-11-26 09:22:02.000000000 +0100
+++ fontconfig-2.10.2/conf.d/Makefile.in	2012-11-26 18:22:38.608844139 +0100
@@ -285,6 +285,9 @@
 	README
 
 CONF_LINKS = \
+	10-antialias.conf \
+	10-hinting.conf \
+	10-hinting-slight.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
@@ -303,7 +306,12 @@
 EXTRA_DIST = $(template_DATA) $(DOC_FILES)
 config_DATA = $(DOC_FILES)
 template_DATA = \
+	10-antialias.conf		\
 	10-autohint.conf		\
+	10-hinting.conf		\
+	10-hinting-full.conf		\
+	10-hinting-medium.conf		\
+	10-hinting-slight.conf		\
 	10-no-sub-pixel.conf		\
 	10-sub-pixel-bgr.conf		\
 	10-sub-pixel-rgb.conf		\
@@ -322,6 +330,7 @@
 	49-sansserif.conf		\
 	50-user.conf			\
 	51-local.conf			\
+	53-monospace-lcd-filter.conf			\
 	60-latin.conf			\
 	65-fonts-persian.conf		\
 	65-khmer.conf			\
