Index: fontconfig-2.10.1/src/fcname.c
===================================================================
--- fontconfig-2.10.1.orig/src/fcname.c	2012-07-24 04:01:32.000000000 +0200
+++ fontconfig-2.10.1/src/fcname.c	2012-08-02 22:28:09.282578437 +0200
@@ -453,6 +453,7 @@
     { (FcChar8 *) "lcddefault",	    "lcdfilter",    FC_LCD_DEFAULT },
     { (FcChar8 *) "lcdlight",	    "lcdfilter",    FC_LCD_LIGHT },
     { (FcChar8 *) "lcdlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
+    { (FcChar8 *) "lcdfilterlegacy",	    "lcdfilter",    FC_LCD_LEGACY },
 };
 
 #define NUM_FC_CONSTANTS   (sizeof _FcBaseConstants/sizeof _FcBaseConstants[0])
