--- /dev/null
+++ b/conf.d/11-lcd-filter-lcddefault.conf
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<!DOCTYPE fontconfig SYSTEM "fonts.dtd">
+<!-- conf.d/lcd-filter-lcddefault.conf -->
+<fontconfig>
+<!--  Use lcddefault as default for LCD filter -->
+  <match target="font">
+    <edit mode="assign" name="lcdfilter">
+      <const>lcddefault</const>
+    </edit>
+  </match>
+</fontconfig>
--- a/conf.d/Makefile.am
+++ b/conf.d/Makefile.am
@@ -37,6 +37,7 @@
 	10-sub-pixel-vbgr.conf \
 	10-sub-pixel-vrgb.conf \
 	10-unhinted.conf \
+	11-lcd-filter-lcddefault.conf \
 	11-lcdfilter-default.conf \
 	11-lcdfilter-legacy.conf \
 	11-lcdfilter-light.conf \
@@ -66,6 +67,7 @@
 	10-antialias.conf \
 	10-hinting.conf \
 	10-hinting-slight.conf \
+	11-lcd-filter-lcddefault.conf \
 	20-fix-globaladvance.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
--- a/conf.d/Makefile.in
+++ b/conf.d/Makefile.in
@@ -288,6 +288,7 @@
 	10-sub-pixel-vbgr.conf \
 	10-sub-pixel-vrgb.conf \
 	10-unhinted.conf \
+	11-lcd-filter-lcddefault.conf \
 	11-lcdfilter-default.conf \
 	11-lcdfilter-legacy.conf \
 	11-lcdfilter-light.conf \
@@ -316,6 +317,7 @@
  	10-antialias.conf \
  	10-hinting.conf \
  	10-hinting-slight.conf \
+	11-lcd-filter-lcddefault.conf \
 	20-fix-globaladvance.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
