--- a/conf.d/Makefile.am
+++ b/conf.d/Makefile.am
@@ -30,6 +30,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
--- a/conf.d/Makefile.in
+++ b/conf.d/Makefile.in
@@ -331,6 +331,7 @@
 	10-hinting.conf \
 	10-hinting-slight.conf \
 	10-scale-bitmap-fonts.conf \
+	11-lcdfilter-default.conf \
 	20-unhint-small-vera.conf \
 	30-urw-aliases.conf \
 	30-metric-aliases.conf \
