Index: fontconfig-2.10.1/conf.d/Makefile.am
===================================================================
--- fontconfig-2.10.1.orig/conf.d/Makefile.am	2012-08-02 22:28:09.318578438 +0200
+++ fontconfig-2.10.1/conf.d/Makefile.am	2012-08-02 22:28:09.334578439 +0200
@@ -41,6 +41,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
Index: fontconfig-2.10.1/conf.d/Makefile.in
===================================================================
--- fontconfig-2.10.1.orig/conf.d/Makefile.in	2012-08-02 22:28:09.318578438 +0200
+++ fontconfig-2.10.1/conf.d/Makefile.in	2012-08-02 22:28:09.338578439 +0200
@@ -276,6 +276,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
