--- a/conf.d/Makefile.am
+++ b/conf.d/Makefile.am
@@ -82,6 +82,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
--- a/conf.d/Makefile.in
+++ b/conf.d/Makefile.in
@@ -332,6 +332,7 @@
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
