--- ../orig/xchat-2.8.8/./src/common/chanopt.c	2009-08-16 09:40:16.000000000 +0000
+++ ./src/common/chanopt.c	2012-09-30 15:30:55.000000000 +0000
@@ -345,7 +345,7 @@
 	}
 }
 
-static void
+static int
 chanopt_save_one_channel (chanopt_in_memory *co, int fh)
 {
 	int i;
@@ -353,10 +353,12 @@
 	guint8 val;
 
 	snprintf (buf, sizeof (buf), "%s = %s\n", "network", co->network);
-	write (fh, buf, strlen (buf));
+	if( write (fh, buf, strlen (buf)) <= 0 )
+		return FALSE;
 
 	snprintf (buf, sizeof (buf), "%s = %s\n", "channel", co->channel);
-	write (fh, buf, strlen (buf));
+	if( write (fh, buf, strlen (buf)) <= 0 )
+		return FALSE;
 
 	i = 0;
 	while (i < sizeof (chanopt) / sizeof (channel_options))
@@ -365,10 +367,13 @@
 		if (val != SET_DEFAULT)
 		{
 			snprintf (buf, sizeof (buf), "%s = %d\n", chanopt[i].name, val);
-			write (fh, buf, strlen (buf));
+			if( write (fh, buf, strlen (buf)) <= 0 )
+				return FALSE;
 		}
 		i++;
 	}
+
+	return TRUE;
 }
 
 void
@@ -377,6 +382,7 @@
 	int i;
 	int num_saved;
 	int fh;
+	ssize_t nb;
 	GSList *list;
 	chanopt_in_memory *co;
 	guint8 val;
@@ -386,12 +392,14 @@
 		return;
 	}
 
-	fh = xchat_open_file ("chanopt.conf", O_TRUNC | O_WRONLY | O_CREAT, 0600, XOF_DOMODE);
+	fh = xchat_open_file ("chanopt.conf.bug147832", O_TRUNC | O_WRONLY | O_CREAT, 0600, XOF_DOMODE);
 	if (fh == -1)
 	{
 		return;
 	}
 
+	nb = 1;
+
 	for (num_saved = 0, list = chanopt_list; list; list = list->next)
 	{
 		co = list->data;
@@ -404,9 +412,11 @@
 			if (val != SET_DEFAULT)
 			{
 				if (num_saved != 0)
-					write (fh, "\n", 1);
+					if( nb > 0 ) nb = write (fh, "\n", 1);
 
-				chanopt_save_one_channel (co, fh);
+				if( nb > 0 )
+					if( chanopt_save_one_channel (co, fh) != TRUE )
+						nb = -1;
 				num_saved++;
 				goto cont;
 			}
@@ -414,13 +424,37 @@
 		}
 
 cont:
+		;
+	}
+
+	if( nb <= 0 )
+	{
+		fprintf( stderr, "chanopt_save_all: write() failed\n" );
+		close( fh );
+		return;
+	}
+
+	if( close (fh) != 0 )
+	{
+		perror( "chanopt_save_all: close() failed" );
+		return;
+	}
+
+	if( xchat_rename_file( "chanopt.conf.bug147832", "chanopt.conf", XOF_DOMODE ) != 0 )
+	{
+		perror( "chanopt_save_all: xchat_rename_file() failed" );
+		return;
+	}
+
+	for (list = chanopt_list; list; list = list->next)
+	{
+		co = list->data;
+
 		g_free (co->network);
 		g_free (co->channel);
 		g_free (co);
 	}
 
-	close (fh);
-
 	/* we're quiting, no need to free */
 
 	/*g_slist_free (chanopt_list);
